app/src/test/java/com/pixlee/pixleesdk/QueryAlbumAPITest.java

```diff
@@ -0,0 +1,199 @@
+        //init album
+        PXLAlbum album = new PXLAlbum(ALBUM_ID, basicDS, analyticsDS);
+        album.setPerPage(40);
+        album.setFilterOptions(fo);
+        album.setSortOptions(so);
+
+        //fire an API
+        Response<PhotoResult> response = album.makeGetAlbumCall().execute();
+        album.setData(response, handlers);
+    }
+
+    @Test
+    public void success() throws Exception {
+        // mock the response
+        intMockServer(
+                HttpURLConnection.HTTP_OK,
+                getAPIJson("albums.from_sku.json")
+        );
+
+        //fileter
+        PXLAlbumFilterOptions fo = new PXLAlbumFilterOptions();
+        fo.minTwitterFollowers = 0;
+        fo.minInstagramFollowers = 0;
+
+        //sort
+        PXLAlbumSortOptions so = new PXLAlbumSortOptions();
+        so.sortType = PXLAlbumSortType.RECENCY;
+        so.descending = true;
+
+        //init album
+        PXLAlbum album = new PXLAlbum(ALBUM_ID, basicDS, analyticsDS);
+        album.setPerPage(40);
+        album.setFilterOptions(fo);
+        album.setSortOptions(so);
+
+        //fire an API
+        Response<PhotoResult> response = album.makeGetAlbumCall().execute();
+        album.setData(response, new PXLBaseAlbum.RequestHandlers<List<PXLPhoto>>() {
+            @Override
+            public void onComplete(List<PXLPhoto> photos) {
+                //success
+                Assert.assertEquals(ALBUM_ID, album.album_id);
+            }
+
+            @Override
+            public void onError(String error) {
+                //failure
+                Assert.fail(error);
+            }
+        });
+    }
+
+    @Test
+    public void success_manyFormats() throws Exception {
+        // mock the response
+        ready(
+                HttpURLConnection.HTTP_OK,
+                getAPIJson("albums.from_sku.many_formats.json"),
+                new PXLBaseAlbum.RequestHandlers<List<PXLPhoto>>() {
+                    @Override
+                    public void onComplete(List<PXLPhoto> photos) {
+                        //success
+                        Assert.assertTrue(photos.size() > 0);
+                    }
+
+                    @Override
+                    public void onError(String error) {
+                        //failure
+                        Assert.fail(error);
+                    }
+                }
+        );
+    }
+
+    @Test
+    public void error_401() throws Exception {
+        // mock the response
+        ready(
+                HttpURLConnection.HTTP_UNAUTHORIZED,
+                getAPIJson("error.401.json"),
+                new PXLBaseAlbum.RequestHandlers<List<PXLPhoto>>() {
+                    @Override
+                    public void onComplete(List<PXLPhoto> photos) {
+                        //fail
+                        Assert.fail();
+                    }
+
+                    @Override
+                    public void onError(String error) {
+                        //an expected case
```

Comment:
Comment spacing is inconsistent in a lot of these files - I think having a space between the `//` and the comment text is nicer, but you can choose whichever you prefer as long as there is consistency in the code base!